### ISSUE TITLE
Problems with ExtractTextPlugin on SCSS

### ISSUE THREAD

**Issue by b12k** [vuejs/vue-loader#51]
Extraction of CSS works absolutely OK.
But when I use "lang='sass'" I get this:
```
ERROR in ./~/extract-text-webpack-plugin/loader.js?{"remove":true}!./~/css-loader!./~/sass-loader!./~/vue-loader/lib/style-rewriter.js!./~/vue-loader/lib/selector.js?type=style&index=0!./src/App/app.vue
Module build failed: TypeError: content.trim is not a function
    at Object.module.exports (D:\OpenServer\domains\test.dev\node_modules\sass-loader\index.js:211:17)
 @ ./src/App/app.vue 1:0-229
```

My webpack.config.js:
```javascript
var webpack = require('webpack'),
    path = require('path'),
    BowerPlugin = require('bower-webpack-plugin'),
    ExtractTextPlugin = require('extract-text-webpack-plugin');

var EXCLUDE = /node_modules|bower_components/;

module.exports = {
    context: path.resolve('src'),
    entry: './index.js',
    output: {
        path: path.resolve('dist'),
        publicPath: '/dist/',
        filename: 'scripts-bundle.min.js'
    },
    module: {
        preLoaders: [
            {
                test: /\.js$/,
                loader: 'eslint',
                exclude: EXCLUDE
            }
        ],
        loaders: [
            {
                test: /\.js$/,
                loader: 'babel',
                exclude: EXCLUDE
            },
            {
                test: /\.scss$/,
                loader: ExtractTextPlugin.extract('style', 'css!autoprefixer!sass'),
                exclude: EXCLUDE
            },
            {
                test: /\.less$/,
                loader: ExtractTextPlugin.extract('style', 'css!autoprefixer!less'),
                exclude: EXCLUDE
            },
            {
                test: /\.css/,
                loader: ExtractTextPlugin.extract('style', 'css!autoprefixer'),
                exclude: EXCLUDE
            },
            {
                test: /\.html$/,
                loader: 'raw',
                exclude: EXCLUDE
            },
            {
                test: /\.vue$/,
                loader: 'vue',
                exclude: EXCLUDE
            }
        ]
    },
    eslint: {
        configFile: path.resolve('.eslintrc')
    },
    babel: {
        optional: ['runtime'],
        loose: 'all',
        nonStandard: false,
        stage: 0
    },
    vue: {
        loaders: {
            css: ExtractTextPlugin.extract('css'),
            scss: ExtractTextPlugin.extract('css!sass')
        }
    },
    autoprefixer: {
        browsers: ['last 2 versions']
    },
    plugins: [
        new BowerPlugin,
        new ExtractTextPlugin('styles-bundle.min.css')
    ],
    watch: true,
    devtool: "source-map"
};
```

Tested css extraction on "vue-loader-example". The same problem. Here is config: 
```javascript
var webpack = require('webpack'),
    ExtractText = require('extract-text-webpack-plugin');

module.exports = {
  entry: './src/main.js',
  output: {
    path: './dist',
    publicPath: '/dist/',
    filename: 'build.js'
  },
  module: {
    loaders: [
      {
        test: /\.vue$/,
        loader: 'vue'
      }
    ]
  },
  vue: {
    loaders: {
      css: ExtractText.extract('css'),
      stylus: ExtractText.extract('css!stylus')
    }
  },
  plugins: [
    new ExtractText('build.css')
  ]
}

if (process.env.NODE_ENV === 'production') {
  module.exports.plugins = [
    new webpack.DefinePlugin({
      'process.env': {
        NODE_ENV: '"production"'
      }
    }),
    new webpack.optimize.UglifyJsPlugin({
      compress: {
        warnings: false
      }
    }),
    new webpack.optimize.OccurenceOrderPlugin()
  ]
} else {
  module.exports.devtool = '#source-map'
}
```
Thanx in advance!

**Comment by dannyfritz** [comment]
I'm having the same problem. Ultimately, I had to remove `vue-loader` until this and #53 are sorted out.

I can't quite put my finger on what the code is doing to transform it from a `.vue` to the 3 separate files, but this is the line i thought was smelly. https://github.com/vuejs/vue-loader/blob/9985f1ff70f612a057cde65bafbde6b8e7c84920/lib/loader.js#L78

**Comment by dannyfritz** [comment]
`vue-loader` is awesome now! thanks

**Comment by dannyfritz** [comment]
I'm actually running into one more problem related to this: source-maps.

#### This works:
```js
less: ExtractTextPlugin.extract('css!less')
```
#### This does not:
```js
less: ExtractTextPlugin.extract('css?sourceMap!less?sourceMap')
```
[`less-loader` and sourcemaps](https://github.com/webpack/less-loader#source-maps)

**Comment by yyx990803** [comment]
Fixed in 0.6.3

**Comment by dannyfritz** [comment]
So it compiles and has a CSS sourcemap now, but it doesn't point to my original LESS code. It points to a compiled version of my Vue style.

I tried adding `<style lang="less?sourceMap">`, but that seems to break the extract-text-plugin and syntax highlighting.

![2015-11-01_13-29-25](https://cloud.githubusercontent.com/assets/315788/10871174/ae4c9610-809c-11e5-9717-5f895037cde9.png)

**Comment by b12k** [comment]
This is awesome! Thank you for an gorgeous framework!

**Comment by yyx990803** [comment]
@dannyfritz fixed again in 6.0.4 :) Should have proper souce map support now

**Comment by dannyfritz** [comment]
@yyx990803 I could shed a tear it is so beautiful. Thanks for the awesome quick turn arounds! :joy:
